security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -307,7 +307,7 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 }
 
 func (sc *SecretController) generateKeyAndCert(saName string, saNamespace string) ([]byte, []byte, error) {
-	id := spiffe.MustGenSpiffeURI(saNamespace, saName)
+	id := spiffe.GenSpiffeURI(saNamespace, saName)
```

Comment:
Please check the returned error here. If an error is returned, the cert generation should fail.